app/components/sidebars/main/channels_list/filtered_list/filtered_list.js

```diff
@@ -294,6 +300,21 @@ class FilteredList extends Component {
             sort(sortChannelsByDisplayName.bind(null, props.intl.locale));
     }
 
+    buildArchivedForSearch = (props, term) => {
+        const {currentChannel, archivedChannels} = props;
+
+        let archived = archivedChannels.map((o) => {
+            return {
+                ...o,
+                fake: true,
+            };
+        });
+        if (!term) { // when there is no search text, display an archived channel only if we are in it.
+            archived = archived.filter((channel) => channel.id === currentChannel.id);
```

Comment:
instead of a map and then a filter you could use a reduce function

Reply:
:+1: